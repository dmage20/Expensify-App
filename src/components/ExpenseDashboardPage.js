import React from 'react';
import ConnectedExpenseList from './ExpenseList';
import ConnectedExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpensesDashboardPage  = () => (
    <div>
      <ExpensesSummary></ExpensesSummary>
      <ConnectedExpenseListFilters />
      <ConnectedExpenseList />
    </div>  
  );

export default ExpensesDashboardPage;