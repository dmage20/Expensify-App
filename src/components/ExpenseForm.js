import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';


export default class ExpenseForm extends React.Component {
    constructor(props) { 
        super(props);

        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            ammount: props.expense? (props.expense.ammount / 100 ).toString() : '',
            createdAt: props.expense? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        }
        console.log(props)
    }
    onDescriptionChange = (e) =>{
        const description = e.target.value
        this.setState(()=>({description}))
    }
    onNoteChange = (e) => {
        const note = e.target.value
        this.setState(()=>({note}))
    }
    onAmmountChange = (e) =>{
        const ammount = e.target.value
        if (!ammount || ammount.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(()=>({ammount}))
        }
    };
    onDateChange = (createdAt) => {
        if (createdAt){
            this.setState(() => ({createdAt}))
        }
    }
    onFocusChange = ({focused}) => {
        this.setState(()=>({calendarFocused: focused}))
    }
    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.ammount || !this.state.description){
            this.setState(() =>({ error: 'Please fill out description and ammount fields' }))
        }else {
            this.setState(() =>({error: ''}))
            this.props.onSubmit({
                description: this.state.description,
                note: this.state.note,
                ammount: parseFloat(this.state.ammount,10)*100,
                createdAt: this.state.createdAt.valueOf()
            })
        }
    }
    render(){
        return (
            <div>
                {this.state.error  && <p>{this.state.error}</p>}
                <form
                    onSubmit={this.onSubmit}
                >
                    <input 
                        type="text" 
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input 
                        type="text" 
                        placeholder="Ammount"
                        value={this.state.ammount}
                        onChange={this.onAmmountChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={ () => false }
                    />
                    <textarea 
                        placeholder="Add a note for the expense (optional)"
                        value={this.state.note}
                        onChange={this.onNoteChange}
                    >
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}
