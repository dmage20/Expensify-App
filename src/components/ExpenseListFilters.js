import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({ startDate, endDate }) =>{
        this.props.setStartDate( startDate );
        this.props.setEndDate( endDate );
    }
    onFocusChange = (calendarFocused) =>{
        this.setState(() => ({calendarFocused}))
    }
    onTextChange = (e)=>{
        this.props.setTextFilter(e.target.value)
    }
    onSortChange = (e) =>{
        if (e.target.value === 'ammount') {
            this.props.sortByAmmount();
        }else if (e.target.value === 'date') {
            this.props.sortByDate();
        }
    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input 
                        type="text" 
                        className="text-input" 
                        placeholder="Search expenses" 
                        value={this.props.filters.text} 
                        onChange={this.onTextChange}
                        />
                    </div>
                    <div className="input-group__item">
                    <select 
                        className="select"
                        value={this.props.filters.sortBy} 
                        onChange={this.onSortChange}>
                            <option value="date">Date</option>
                            <option value="ammount">Ammount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker 
                        startDate={this.props.filters.startDate}
                        endDate={this.props.filters.endDate}
                        onDatesChange={this.onDatesChange}
                        showClearDates={true}
                        focusedInput={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                        />
                    </div>
                </div>



            </div>
        )
    }
}

const mapStateToProps = (state) => ({
        filters: state.filters
    });
const mapDispatchtoProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByAmmount: () => dispatch(sortByAmmount()),
    sortByDate: () => dispatch(sortByDate())

})

export default connect(mapStateToProps, mapDispatchtoProps)(ExpenseListFilters);