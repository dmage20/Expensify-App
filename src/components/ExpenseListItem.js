import React from 'react';
import {Link} from 'react-router-dom';

const ExpenseListItem = ({ ammount , description, createdAt, id}) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
            <p>$ {ammount} date: {createdAt}</p>
        
    </div>
);


export default ExpenseListItem;