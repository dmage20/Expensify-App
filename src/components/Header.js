
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/authentication';

export const Header = ({startLogout}) => (
    <div>
        <h1>Expensify</h1>
        <NavLink activeClassName="is-active" to="/" exact={true}>Home</NavLink>
        <NavLink activeClassName="is-active" to="/dashboard" exact={true}>Dashboard</NavLink>
        <NavLink activeClassName="is-active" to="/create">Create Expense</NavLink>,
        <NavLink activeClassName="is-active" to="/help">Help</NavLink>
        <button onClick={startLogout}>Logout</button>
    </div>
);


const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})
export default connect(undefined, mapDispatchToProps)(Header)