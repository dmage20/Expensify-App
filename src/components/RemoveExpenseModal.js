import React  from 'react';
import Modal from 'react-modal';


const RemoveExpenseModal = ({selectedExpense, onRemoveConfirm, handleCancelRemove})=>{
    
    return(
        <Modal
            ariaHideApp={false}
            isOpen={!!selectedExpense}
            onRequestClose={handleCancelRemove}
            constentLabel="Remove this expense?"
            coloseTimeoutMS={200}
            className="modal"
        >
            <h3 className="modal__title">Remove this expense?</h3>
            <button className="button" 
            onClick={onRemoveConfirm}>Remove</button>

        </Modal>
)};

export default RemoveExpenseModal;