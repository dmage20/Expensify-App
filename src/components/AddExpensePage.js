import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {addExpnese} from '../actions/expenses';


export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpnese(expense);
        this.props.history.push('/')
    };
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}
const mapDispatchToProps = (dispatch) =>( {
    addExpnese: (expense) => dispatch(addExpnese(expense))
})
export default connect(undefined, mapDispatchToProps)(AddExpensePage);