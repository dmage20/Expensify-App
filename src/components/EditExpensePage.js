import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import RemoveExpenseModal from './RemoveExpenseModal';

export class EditExpensePage extends React.Component {
   state = {
    selectedExpense: undefined
   }
  
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onStartRemove = () => {
    // this.props.startRemoveExpense({ id: this.props.expense.id });
    // this.props.history.push('/');
    this.setState( () => ({selectedExpense: this.props.expense}));

  };
  onRemoveConfirm = () =>{
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  }
  handleCancelRemove = () => {
    this.setState({selectedExpense: undefined})
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
            <button className="button button--secondary" onClick={this.onStartRemove}>Remove Expense</button>
        </div>
        <RemoveExpenseModal selectedExpense={this.state.selectedExpense} onRemoveConfirm={this.onRemoveConfirm} handleCancelRemove={this.handleCancelRemove} />
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);