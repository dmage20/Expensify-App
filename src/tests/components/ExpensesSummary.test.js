import React from 'react';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import {shallow} from 'enzyme';

test('should display one expense', () =>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={195}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should display multiple expenses', () =>{
    const wrapper = shallow(<ExpensesSummary expenseCount={12} expensesTotal={195900}/>)
    expect(wrapper).toMatchSnapshot()
})