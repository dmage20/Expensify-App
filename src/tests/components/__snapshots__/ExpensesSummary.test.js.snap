// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display multiple expenses 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={12}
    expensesTotal={195900}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Viewing 
        12
         
        expenses
         totaling 
        $1,959.00
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          12,
          " ",
          "expenses",
          " totaling ",
          "$1,959.00",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        12,
        " ",
        "expenses",
        " totaling ",
        "$1,959.00",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Viewing 
          12
           
          expenses
           totaling 
          $1,959.00
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            12,
            " ",
            "expenses",
            " totaling ",
            "$1,959.00",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          12,
          " ",
          "expenses",
          " totaling ",
          "$1,959.00",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should display one expense 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={1}
    expensesTotal={195}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Viewing 
        1
         
        expense
         totaling 
        $1.95
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totaling ",
          "$1.95",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        1,
        " ",
        "expense",
        " totaling ",
        "$1.95",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Viewing 
          1
           
          expense
           totaling 
          $1.95
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            1,
            " ",
            "expense",
            " totaling ",
            "$1.95",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totaling ",
          "$1.95",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
